### PR TITLE
feat(lineage): add time filtering for lineage edges

### PR DESCRIPTION
**Pull request by aditya-radhakrishnan**
## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [x] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Comment by jjoyce0510** [comment]
metadata-io failing build

File: metadata-jobs/mae-consumer/src/main/java/com/linkedin/metadata/kafka/hook/UpdateIndicesHook.java [hunk 1]
```diff
@@ -364,12 +371,39 @@ private void updateGraphServiceDiff(
...
+    if (mergedEdges.size() > 0) {
+      log.debug("Updating edges: {}", mergedEdges);
+      mergedEdges.forEach(_graphService::upsertEdge);
+    }
+  }
+
+  private static List<Edge> getMergedEdges(final Set<Edge> oldEdgeSet, final Set<Edge> newEdgeSet) {
+    final Map<Integer, com.linkedin.metadata.graph.Edge> oldEdgesMap = oldEdgeSet
+        .stream()
+        .map(edge -> Pair.of(edge.hashCode(), edge))
```
Comment (jjoyce0510) [inline metadata-jobs/mae-consumer/src/main/java/com/linkedin/metadata/kafka/hook/UpdateIndicesHook.java:385]: this could bite us later

**Review by jjoyce0510 — APPROVED** [review]
Pretty rad change. Once we go green I think we're good to go. Given the size of this change, let's be very alert w.r.t. what we hear frm community

File: metadata-jobs/mae-consumer/src/main/java/com/linkedin/metadata/kafka/hook/UpdateIndicesHook.java (hunk 1 above)
Comment (aditya-radhakrishnan) [reply metadata-jobs/mae-consumer/src/main/java/com/linkedin/metadata/kafka/hook/UpdateIndicesHook.java:385]: like surround it with a try/catch?

Or wdym?

**Pull request merged**